src/algorithms/digi/LGADPulseDigitization.cc

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Souvik Paul, Chun Yuen Tsang, Prithwish Tribedy
+// Special Acknowledgement: Kolja Kauder
+//
+// Convert ADC pulses from LGADPulseGeneration into ADC and TDC values
+
+#include <podio/RelationRange.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <cmath>
+#include <gsl/pointers>
+#include <limits>
+#include <vector>
+
+#include "LGADPulseDigitization.h"
+#include "algorithms/digi/LGADPulseDigitizationConfig.h"
+
+namespace eicrecon {
+
+void LGADPulseDigitization::process(const LGADPulseDigitization::Input& input,
+                                   const LGADPulseDigitization::Output& output) const {
+  const auto [simhits] = input;
+  auto [rawhits]       = output;
+
+  double thres = -m_cfg.t_thres;
+  int adc_range = m_cfg.adc_range;
+
+  for (const auto& pulse : *simhits) {
+    double intersectionX = 0.0;
+    int tdc              = std::numeric_limits<int>::max();
+    int adc              = 0;
+    double V             = 0.0;
+
+    int time_bin         = 0;
+    double adc_prev      = 0;
+    double time_interval = pulse.getInterval();
+    auto adcs            = pulse.getAdcCounts();
+    double n_EICROC_cycle = static_cast<int>(std::floor(pulse.getTime()/m_cfg.tMax + 1e-3));
+    for (const auto adc : adcs) {
+      if (adc_prev >= thres && adc <= thres) {
+        intersectionX = time_bin * time_interval +
+                        time_interval * (thres - adc_prev) / (adc - adc_prev);
+        tdc = static_cast<int>(intersectionX / time_interval) + n_EICROC_cycle * m_cfg.tdc_range;
```

Comment:
```suggestion        
        tdc = time_bin + n_EICROC_cycle * m_cfg.tdc_range;
```
I believe the code is currently doing anyway. The electronics cannot make use of the intersectionX as it doesn't have access to the adc and adc_prev values, only that a threshold was crossed at some point in the time bin.